boards/arduino-common/board.c

```diff
@@ -59,7 +59,7 @@ void board_init(void)
 void SystemInit(void)
 {
     /* initialize UART_0 for use as stdout */
-    uart_stdio_init();
+    stdio_init();
 
     stdout = &uart_stdout;
```

Comment:
Are you referring to the comments above?  The only reason I left them is because the arduino board is dependent on the atmega_common CPU which sets the stdio to UART.  There is no ability to override.  Maybe we should also allow it to be overridable while were at it?